src/hiscores.ts

```diff
@@ -64,8 +65,14 @@ export async function getOfficialStats(
   try {
     const response = await httpGet<HiscoresResponse>(url, config);
     return response.data;
-  } catch {
-    throw Error(PLAYER_NOT_FOUND_ERROR);
+  } catch (err) {
+    const error = err as Error | AxiosError;
+
+    if (!axios.isAxiosError(error)) throw err;
+
+    if (error.response?.status === 404) throw new PlayerNotFoundError();
```

Comment:
Instead of typecasting error as `Error | AxiosError` above, you can change the condition here to `error instanceof Error && error.response?.status === 404`

Reply:
I misinterpreted the code here, resolving this in favor of a new comment.